pykern/pkcli/__init__.py

```diff
@@ -66,6 +66,41 @@ def command_error(fmt, *args, **kwargs):
     raise CommandError(fmt.format(*args, **kwargs))
 
 
+class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
+                      argparse.RawDescriptionHelpFormatter):
+    def _expand_help(self, action):
+        r = super()._expand_help(action)
+        return r.split('\n')[0]
+
+class CustomParser(argparse.ArgumentParser):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.program = { key: kwargs[key] for key in kwargs }
+        self.options = []
+
+    def format_help(self):
+        formatter = argh.PARSER_FORMATTER(prog=self.prog)
+        if not self.description:
+            formatter = CustomFormatter(prog=self.prog)
+        formatter.add_usage(self.usage, self._actions,
+                            self._mutually_exclusive_groups)
+        formatter.add_text(self.description)
+        for action_group in self._action_groups:
+            formatter.start_section(action_group.title)
+            formatter.add_text(action_group.description)
+            formatter.add_arguments(action_group._group_actions)
+            formatter.end_section()
+        formatter.add_text(self.epilog)
+        return formatter.format_help()
+
+    def print_help(self):
+        h = self.format_help()
+        if not self.description:
```

Comment:
Perhaps it would be better to just do this in format_help since you know if there's a description or not.